Improved Weather Data Handling

Amansingh0807 opened the pull request:
Implemented the rendering of a 7-day weather forecast using the fetched data, or remove if not necessary. Added a default placeholder icon if icon_url is not available for current weather conditions. Added units for both wind speed (m/s) and humidity (%), making it clear for users. Ensured that the necessary CSS classes (.city-name, .temp, .temp-icon, etc.) are defined and used to make the app visually appealing.

I did all the improvements as suggested in the issue number #4

Amansingh0807 committed:
Improved Weather Data Handling

Implemented the rendering of a 7-day weather forecast using the fetched data, or remove if not necessary.
Added a default placeholder icon if icon_url is not available for current weather conditions.
Added units for both wind speed (m/s) and humidity (%), making it clear for users.
Ensured that the necessary CSS classes (.city-name, .temp, .temp-icon, etc.) are defined and used to make the app visually appealing.

I did all the improvements as suggested in the issue number #4

yatikakain commented:
@Amansingh0807 can you share the final ui screenshot for 7 day forecast that you implemented?

Amansingh0807 commented:
> Implemented the rendering of a 7-day weather forecast using the fetched data, or remove if not necessary. Added a default placeholder icon if icon_url is not available for current weather conditions. Added units for both wind speed (m/s) and humidity (%), making it clear for users. Ensured that the necessary CSS classes (.city-name, .temp, .temp-icon, etc.) are defined and used to make the app visually appealing.

Hey @yatikakain i was working on your project thoroughly I noticed there are some other things too which are needed to improve. before adding CSS to the forecast 7-days it will look like this :--


![Screenshot 2025-01-09 143005](https://github.com/user-attachments/assets/96b7b928-8f77-47d5-b2fd-92d628e494bf)
![Screenshot 2025-01-09 143017](https://github.com/user-attachments/assets/9e7c59e4-3ca1-4084-a7b2-d77bf80be583)



But after adding styling to the component it will look like this :--
![Screenshot 2025-01-09 143232](https://github.com/user-attachments/assets/8b16b830-e0a6-457f-b7ba-226b26d1bdc1)
![Screenshot 2025-01-09 143239](https://github.com/user-attachments/assets/d5fa832f-5730-4b4a-8cee-2932d7cdacd8)



I am still working on the same, Improving styling specially the credit part and also the button of get weather when they will be completed i will Push all the code into the forked repository of mine and will add those changes into this pull request 

and also i used open weather API key in this 
as may be in future if you wish to scale and make feature-rich weather service which includes a production-grade application, historical records, air pollution, or maps for all of these features i included open weather API but if you want i will do it by she codes weather API too 


Thank You 
best regards,
Aman Singh

yatikakain commented:
@Amansingh0807  Yes, please fix the UI, and try to maintain the current UI style and theme as much as possible.

Amansingh0807 commented:
> es, please fix the UI, and try to maintain the current UI style and theme as much as possible.

Ok i will remember this

Amansingh0807 commented:
> > es, please fix the UI, and try to maintain the current UI style and theme as much as possible.
> 
> Ok i will remember this

BTW whats your take on using open weather api ?
should i implement that or use she codes weather api ?

yatikakain commented:
@Amansingh0807 we can use it in future, but for now I think shecodes api is working fine.
